### ISSUE TITLE
Indium incompatible with sodium 0.5.9 

### ISSUE THREAD

**Issue by Neodious** [comp500/Indium#307]
I'm on  Minecraft 1.20.6 and Indium is incompatible  with sodium 0.5.9 but I'm getting a crash error when using sodium 0.5.8 so need to use sodium 0.5.9 but Indium isn't supported by that version so i can't use Distance Horizons as it needs Indium to work anything i can do or do i have to wait for Indium to update to support sodium 0.5.9

**Comment by Neodious** [comment]
[crash-2024-06-15_01.21.16-client.txt](https://github.com/user-attachments/files/15848463/crash-2024-06-15_01.21.16-client.txt)

**Comment by Neodious** [comment]
---- Minecraft Crash Report ----
// My bad.

Time: 2024-06-15 01:21:16
Description: Unexpected error

java.lang.RuntimeException: Mixin transformation of me.jellysquid.mods.sodium.client.render.immediate.CloudRenderer failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at net.minecraft.class_761.method_3259(class_761.java:11699)
	at net.minecraft.class_761.method_22710(class_761.java:1294)
	at net.minecraft.class_757.method_3188(class_757.java:1093)
	at net.minecraft.class_757.method_3192(class_757.java:858)
	at net.minecraft.class_310.method_1523(class_310.java:1320)
	at net.minecraft.class_310.method_1514(class_310.java:888)
	at net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422)
	... 14 more
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [mixins.iris.compat.sodium.json:clouds.MixinCloudRenderer from mod iris] from phase [DEFAULT] in config [mixins.iris.compat.sodium.json] FAILED during APPLY
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:638)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:589)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:379)
	... 17 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Critical injection failure: @Inject annotation on buildIrisVertexBuffer could not find any targets matching 'render(Lnet/minecraft/class_638;Lnet/minecraft/class_746;Lnet/minecraft/class_4587;Lorg/joml/Matrix4f;FFDDD)V' in me.jellysquid.mods.sodium.client.render.immediate.CloudRenderer. No refMap loaded. [PREINJECT Applicator Phase -> mixins.iris.compat.sodium.json:clouds.MixinCloudRenderer from mod iris -> Prepare Injections ->  -> handler$bip000$iris$buildIrisVertexBuffer(Lnet/minecraft/class_638;Lnet/minecraft/class_746;Lnet/minecraft/class_4587;Lorg/joml/Matrix4f;FFDDDLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V -> Parse]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.validateTargets(InjectionInfo.java:658)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.findTargets(InjectionInfo.java:589)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.readAnnotation(InjectionInfo.java:330)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.<init>(InjectionInfo.java:316)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.<init>(InjectionInfo.java:308)
	at org.spongepowered.asm.mixin.injection.struct.CallbackInjectionInfo.<init>(CallbackInjectionInfo.java:46)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo$InjectorEntry.create(InjectionInfo.java:149)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.parse(InjectionInfo.java:710)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.prepareInjections(MixinTargetContext.java:1329)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.prepareInjections(MixinApplicatorStandard.java:1053)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:327)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:422)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 17 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at net.minecraft.class_761.method_3259(class_761.java:11699)
	at net.minecraft.class_761.method_22710(class_761.java:1294)
	at net.minecraft.class_757.method_3188(class_757.java:1093)

-- Uptime --
Details:
	JVM uptime: 26.471s
	Wall uptime: 18.462s
	High-res time: 17.156s
	Client ticks: 220 ticks / 11.000s
Stacktrace:
	at net.minecraft.class_310.method_1587(class_310.java:2548)
	at net.minecraft.class_310.method_54580(class_310.java:983)
	at net.minecraft.class_310.method_1514(class_310.java:911)
	at net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Affected level --
Details:
	All players: 1 total; [class_746['OGNeodious'/36, l='ClientLevel', x=-7.50, y=78.00, z=4.50]]
	Chunk stats: 1024, 371
	Level dimension: minecraft:overworld
	Level spawn location: World: (0,77,0), Section: (at 0,13,0 in 0,4,0; chunk contains blocks 0,-64,0 to 15,319,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 1308 game time, 1308 day time
	Server brand: fabric
	Server type: Integrated singleplayer server
	Tracked entity count: 81

**Comment by Neodious** [comment]
That's the error I get when I load the game with Indium 1.0.31 for 1.20.6 and sodium 0.5.8 if anyone can help that would be great

**Comment by SybeDB** [comment]
I have noticed this same issue when I run indium with sodium. I have found that running iris with any sort of shader (that is not too minimal (for example onlywater doesn't work)) fixes the issue.

Edit:

So, I got the same error even without indium, just using iris and sodium. I used Complementary reimagined 5.1.1. It's weird because complementary reimagined 2.0.3 does work so there is apparently an important difference.
